commons/src/main/java/org/infinispan/commons/dataconversion/MediaType.java

```diff
@@ -121,14 +129,6 @@ String getTree() {
       return toString();
    }
 
-   void setTree(String tree) {
-      MediaType type = MediaType.fromString(tree);
-      this.type = type.type;
-      this.subType = type.subType;
-      this.typeSubtype = type.typeSubtype;
-      this.weight = type.weight;
-   }
-
    public static MediaType fromString(String mediaType) {
```

Comment:
I see ```fromString``` is just an alias for ```parse```. I'd keep just ```fromString``` , deprecate ```parse``` and replace all its usages.